### ISSUE TITLE
Issues with CollisionDetector and BallPaths

### ISSUE THREAD

**Issue by brandonLi8** [phetsims/collision-lab#75]
Currently, our physics engine works by first detecting if there is overlap between balls or a ball and the border, then calculating how long they have been overlapping, then reconstructing the collision and placing the ball at where it would be at the current time. This works great. However, in #61, now that there are paths, which works by recording the position of the ball, it becomes obvious (particularly with high velocity balls).

For instance,
![image](https://user-images.githubusercontent.com/42391580/83049648-bda34d80-a008-11ea-8e45-ca2a71e0fa49.png)

the blue ball didn't collide twice here, as it may appear based on the path. It actually just bounced off the wall, yet there are two cusps in the path. Instead, at the first corner, the ball isn't yet overlapping with the ball. Then on the next frame, the ball is overlapping with the ball by a considerate amount. The physics engine reconstructs and places the ball where it would be if it where real life, at the second cusp. But the path doesn't record where the ball actually hit the wall and at what time.

With, slower speed collisions, this isn't as visible, but it is still there. When i change the thickness of the line to make it more visible it looks like
![image](https://user-images.githubusercontent.com/42391580/83050170-5fc33580-a009-11ea-83bc-46ca60c10131.png)

This happens with both ball to ball and ball to wall collisions:
![image](https://user-images.githubusercontent.com/42391580/83050270-8aad8980-a009-11ea-90f1-dfac73a5a08e.png)

**Comment by brandonLi8** [comment]
As far as I am aware, this is also an issue in the flash version. Although, the flash version uses dotted lines, so it is harder to spot.
![image](https://user-images.githubusercontent.com/42391580/83050565-fc85d300-a009-11ea-8bce-bdd043ff993d.png)

**Comment by brandonLi8** [comment]
In the commit above, I fixed this issue.

In the collision detector, I calculated the position that the ball and wall or ball to ball collided and recorded that point at the appropriate time. Now the path is guaranteed to have the spot where balls are colliding with a wall or with another ball (in the correct position, not the position of overlap).

This results look really good.
![image](https://user-images.githubusercontent.com/42391580/83168533-daa25400-a0ce-11ea-9a46-68df5f4adc23.png)
(super high velocity ball)

![image](https://user-images.githubusercontent.com/42391580/83168569-e68e1600-a0ce-11ea-9a6e-dfab5d757464.png)

(two high velocity balls colliding)
